xbmc/windowing/Resolution.cpp

```diff
@@ -211,14 +225,28 @@ void CResolutionUtils::FindResolutionFromWhitelist(float fps, int width, int hei
           (info.dwFlags & D3DPRESENTFLAG_MODEMASK) == (curr.dwFlags & D3DPRESENTFLAG_MODEMASK) &&
           MathUtils::FloatEquals(info.fRefreshRate, fps * 2.5f, 0.01f))
       {
-        CLog::Log(
-            LOGDEBUG,
-            "[WHITELIST] Matched an exact resolution with a 3:2 pulldown refresh rate {} ({})",
-            info.strMode, i);
-        resolution = i;
-        return;
+        float fit_ratio = (height == info.iScreenHeight)
+                              ? static_cast<float>(width) / static_cast<float>(info.iScreenWidth)
+                              : static_cast<float>(height) / static_cast<float>(info.iScreenHeight);
+        if (fit_ratio > 1.0f)
+          fit_ratio = 1.0f / fit_ratio;
+
```

Comment:
That's the duplication from above. It won't work in most cases. AR needs to come from videoplayer.

Reply:
The same. It is not about AR.